com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JavaHotCodeReplaceProvider.java

```diff
@@ -300,14 +303,21 @@ public void onClassRedefined(Consumer<List<String>> consumer) {
 
     @Override
     public CompletableFuture<List<String>> redefineClasses() {
+        JobHelpers.waitForBuildJobs(10 * 1000);
         return CompletableFuture.supplyAsync(() -> {
             List<String> classNames = new ArrayList<>();
+            boolean success;
             synchronized (this) {
                 classNames.addAll(deltaClassNames);
-                doHotCodeReplace(deltaResources, deltaClassNames);
+                success = doHotCodeReplace(deltaResources, deltaClassNames);
                 deltaResources.clear();
                 deltaClassNames.clear();
             }
+
+            if (!classNames.isEmpty() && !success) {
```

Comment:
What if `classNames` is empty? (I imagine it's the case of clicking HCR without any code changes, repeatedly). Does it return successfully without a warning?

Reply:
The client will show a message saying no changed classes.